src/api/authentication.js

```diff
@@ -164,15 +73,15 @@ function isAuthenticationTypeEnabled(type) {
   return getEnabledAuthenticationTypesFromConfig(config).includes(type)
 }
 
-async function authenticateRequest(ctx) {
-  let user
+async function authenticateRequest(ctx, next) {
+  let user = null
   // First attempt basic authentication if enabled
   if (user == null && isAuthenticationTypeEnabled('basic')) {
-    user = await authenticateBasic(ctx)
+    user = await authenticateBasic(ctx, next)
   }
   // Otherwise try token based authentication if enabled
-  if (user == null && isAuthenticationTypeEnabled('token')) {
-    user = await authenticateToken(ctx)
+  if (user == null && isAuthenticationTypeEnabled('local') && ctx.req.user) {
```

Comment:
is ctx.req.user a reference to the req.session object ? or is there a middleware that performs "ctx.req.user = getUserbySessionId(req.session)"